lib/hash.rb

```diff
@@ -20,15 +20,10 @@ def method_missing(method, *args)
 
     if setter?(method)
       self[prop] = args.first
-    elsif key?(prop)
+    elsif key?(prop) || hash_dot_use_default
```

Comment:
I think the tests will catch this once they're updated, but this currently only uses the instance level `hash_dot_use_default` setting. Seems like we should come up with a method like `[to_dot?](https://github.com/adsteel/hash_dot/blob/master/lib/hash.rb#L42)` that checks both instance and class level settings.